app/src/main/java/de/westnordost/streetcomplete/quests/crossing_type/AddCrossingTypeForm.java

```diff
@@ -0,0 +1,43 @@
+package de.westnordost.streetcomplete.quests.crossing_type;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.quests.ImageListQuestAnswerFragment;
+
+public class AddCrossingTypeForm extends ImageListQuestAnswerFragment
+{
+    private final OsmItem[] TYPES = new OsmItem[] {
+            new OsmItem("traffic_signals", R.drawable.crossing_type_signals, R.string.quest_crossing_type_signals),
+            new OsmItem("uncontrolled", R.drawable.crossing_type_zebra, R.string.quest_crossing_type_uncontrolled),
+            new OsmItem("island", R.drawable.crossing_type_island, R.string.quest_crossing_type_island),
+            new OsmItem("unmarked", R.drawable.crossing_type_unmarked, R.string.quest_crossing_type_unmarked)
+    };
+
+    @Override protected int getMaxSelectableItems()
+    {
+        return -1;
+    }
+
+    @Override protected OsmItem[] getItems()
+    {
+        return TYPES;
+    }
+
+    @Override protected int getItemsPerRow()
+    {
+        return 4;
```

Comment:
Do you like that?
<img src="https://user-images.githubusercontent.com/25306497/28823684-20b78b64-76bf-11e7-96f5-e1fd38af7b0a.png" width="300px">